### DISCUSSION TITLE
CSV out for report?

### DISCUSSION THREAD

**Discussion by austimkelly** [praetorian-inc/noseyparker#123] [Q&A]
Hi  -I see json, jsonl, and SARIF. Is there CSV output I missed?
Thanks for the tool, it's blazing fast!

**Comment by bradlarsen** [comment]
You're welcome!

There is no CSV output at present. It's a little bit tricky to represent the finding data as flat CSV, since it's nested: each individual finding has 1+ occurrences associated with it, and each of those occurrences has 0+ provenance entries associated with it.

If you wanted to produce a CSV with just some of the information, you could write a JSON -> CSV translator script, such as with Python or [jq](https://github.com/jqlang/jq).

What columns would you want in hypothetical CSV output?

**Reply by austimkelly** [reply]
Hi Brad - Thanks for your reply. I'm just trying to get the repo name and original source file.
Here's my hacky solution to read out of provenance: https://github.com/austimkelly/secretsynth/blob/main/org-scan/noseyparker_scan.py#L36
So really just the sink information (as in sources and sinks). Cheers, --Tim
